File: RadStudio_XE6_Demo/radstudiodemos-code-1932-branches-RadStudio_XE6/Object Pascal/DataSnap/connectors_many_devicies/WindowsPhone7Clients/CompanyTweetClient/CompanyTweetClient/CTCallback.cs
using Embarcadero.Datasnap.WindowsPhone7;
using Microsoft.Phone.Controls;
using System.Windows;
using System.Threading;
using Microsoft.Xna.Framework.Audio;
using Microsoft.Xna.Framework;
using System.Collections.Generic;
using System.Collections.ObjectModel;


namespace CompanyTweetClient
{
    public class CTCallback : DBXCallback
    {
        private SynchronizationContext Context;
        public CTCallback(SynchronizationContext contxt)
            : base()
        {
            Context = contxt;
        }

        public override TJSONValue Execute(TJSONValue value, int JSONType)
        {
            TJSONObject obj = TJSONObject.Parse(value.ToString());
            Context.Send(new SendOrPostCallback(x =>
                {
                    PhoneApplicationFrame frame = Application.Current.RootVisual as PhoneApplicationFrame;
                    PhoneApplicationPage mp = frame.Content as PhoneApplicationPage;
                    if (mp is MainTweetPage)
                    {
                        ((MainTweetPage)mp).listBox1.Items.Add(obj.getString("username") + ": " + obj.getString("message"));
                        SoundEffect.MasterVolume = 1.0f;
                        SoundEffect sfx = SoundEffect.FromStream(TitleContainer.OpenStream("tweet.wav"));
                        SoundEffectInstance sfxInstance = sfx.CreateInstance();
                        sfxInstance.Play();
                    }
                }), null);
            return null;
        }
    }
}

File: RadStudio_XE6_Demo/radstudiodemos-code-1932-branches-RadStudio_XE6/Object Pascal/DataSnap/connectors_many_devicies/AndroidClients/AndroidDemoClientApp/src/com/embarcadero/androiddemoclientapp/CTCallback.java
package com.embarcadero.androiddemoclientapp;

import android.os.Bundle;
import android.os.Handler;
import android.os.Message;
import android.util.Log;

import com.embarcadero.javaandroid.DBXCallback;
import com.embarcadero.javaandroid.TJSONObject;
import com.embarcadero.javaandroid.TJSONValue;

public class CTCallback extends DBXCallback {

	private Handler handler;

	public CTCallback(Handler handler) {
		super();
		this.handler = handler;
	}

	@Override
	public TJSONValue execute(TJSONValue value, int JSONType) {
		Log.i("callback", value.toString());
		TJSONObject obj =  (TJSONObject) value;
		Message m = handler.obtainMessage(1); //it's a simple tweet message
		Bundle b = new Bundle();
		b.putString("username", obj.getString("username"));
		b.putString("message", obj.getString("message"));
		m.setData(b);
		handler.sendMessage(m);
		return null;
	}

}
